dask/dataframe/partitionquantiles.py

```diff
@@ -383,6 +383,9 @@ def process_val_weights(vals_and_weights, npartitions, dtype_info):
     elif "datetime64" in str(dtype):
         rv = pd.DatetimeIndex(rv, dtype=dtype)
     elif rv.dtype != dtype:
+        if is_integer_dtype(dtype):
+            # pandas EA raises instead of truncating
```

Comment:
Do you an example of the error raised here you can share?

Reply:
```
df = DataFrame(
    {
        "a": 1,
        "b": Series([3, 1, 2], dtype="Int64"),
    }
)
dd.from_pandas(df, npartitions=3).set_index("b")
```

if partitions+1 is larger than the number of rows